apple/Tests/Utils/RefTest.swift

```diff
@@ -0,0 +1,23 @@
+import CustomDump
+import Foundation
+import Sargon
+import SargonUniFFI
+import XCTest
+
+class RefTest<SUT_: SargonReferenceType>: Test<SUT_> {
+	func test_take_twice_throws() {
+		func doTest(_ sut: SUT) {
+			XCTAssertNoThrow(try sut.take())
+			XCTAssertThrowsError(try sut.take())
```

Comment:
`take()` **consumes** `inner` **without making any expensive `clone()`**, and will throw if call a second time.